### ISSUE TITLE
command to open Jupyter notebook as tab downloads it instead

### ISSUE THREAD

**Issue by rasbt** [jupyterlab/jupyterlab#5380]
**Describe the bug**

I am 95% sure that this is a bug. In particular, I am trying to open multiple Jupyter notebooks as "tabs" but instead, the browser (latest Safari, v 12.0) download the notebook instead of opening it.

**To Reproduce**

1. Open a new Jupyter Lab session
2. Go to the Files browser in Jupyter Lab's webbrowser view
3. Right-click on a .ipynb file and click on "open in new browser tab"

Instead of opening the notebook, it's downloaded to the download folder.

**Expected behavior**

A new browser tab opening with the selected notebook

**Screenshots**

![screen shot 2018-09-25 at 1 17 04 pm](https://user-images.githubusercontent.com/5618407/46034309-eea9dd00-c0c5-11e8-9fc7-a73a6cc94420.png)

**Desktop (please complete the following information):**

- OS: macOS 10.14
- Browser: Safari 12.0
- JupyterLab 0.34.9

**Comment by skane88** [comment]
Same issue appears to be occurring on Chrome - right click on the notebook, select Open in New Browser Tab and the browser's download file window pops up. Running JupyterLab version 0.34.9 (installed via Conda) and Chrome 69.0.3497.100

**Comment by ian-r-rose** [comment]
This is the intended behavior. This opens the contents of the file in a new browser tab (but *not* in a new JupyterLab workspace). If the browser has a native renderer for that file type (e.g., an image or PDF) it should render that. It does not, however, have a notebook renderer, so it downloads the file.

If you want to open the file in a new tab, you need to open a new JupyterLab workspace and open that file.

Clearly, the naming of this menu item is a bit confusing. How do you think we can make it more obvious what it does?

**Comment by rasbt** [comment]
Oh I see! Makes sense then -- I just started using Jupyter Lab (coming from Jupyter Notebook) and somehow had this assumption that multiple Jupyter notebooks could be opened in the same session (thought I've seen multiple tabs at a presentation once or read about it or sth like that).

Regarding the naming, I can't think of a better name right now that wouldn't be too lengthy (Open in New Browser Tab Outside this Workspace). One could display a warning message saying that the item is not opened inside the current session, but this could quickly become annoying. 

Honestly, I am not sure how useful this current "Open in New Browser Tab" feature/link is, and maybe it doesn't have to be shown that prominently. I.e., I would say it would make sense to also move it into the 
"Open with" menu. Right now, we have 

Open with:
- Notebook
- Editor
- JSON

You could add a "Web Browser" or "Browser Tab" entry to it?

I would argue that any of the three (Notebook, Editor, JSON) is more useful than opening some file in my browser tab (because that's not what people are using Jupyter Lab for). So, having "Open in New Browser Tab" as top-level menu bar entry vs the items in Open With is maybe the cause for the confusion (i.e., because it's shown so prominently, people assume that it's some Jupyter Lab feature).

**Comment by paulperry** [comment]
I landed here due to the same confusion.  I could not figure out how to open a second view of a notebook and I knew it could be done, till I found this menu in File: "New view for Notebook".  I think it's more obvious to put it off the menu of the notebook itself as "Open New View" .  

The context menu off of the notebook in the file view also has an "Open" menu item with a folder icon that doesn't do anything, at least for me.

**Comment by jasongrout** [comment]
> The context menu off of the notebook in the file view also has an "Open" menu item with a folder icon that doesn't do anything, at least for me.

That will open the file if it isn't already open, and will activate the existing tab if it is already open.

**Comment by jasongrout** [comment]
> > I would argue that any of the three (Notebook, Editor, JSON) is more useful than opening some file in my browser tab (because that's not what people are using Jupyter Lab for). So, having "Open in New Browser Tab" as top-level menu bar entry vs the items in Open With is maybe the cause for the confusion (i.e., because it's shown so prominently, people assume that it's some Jupyter Lab feature).
> 
> That's a good point @rasbt. @tgeorgeux do you have any insight into how prominent this action should be vs. other open actions?

I think this was confusing to at least one other person in #7900. Revisiting this: @tgeorgeux - thoughts?

On the one hand, "Open with..." is really, right now, "Open in JupyterLab with...", while the sibling "Open in New Browser Tab" is really similar to what GitHub calls the "raw" file view, so perhaps something like "Open as Raw File in New Browser Tab" would be more clarifying (though too long), or "Open as Raw File" or something?

**Comment by RedTn** [comment]
People seem to be using this button assuming it's a "open in a new jupyterlab workspace", would it be ok to change to that?
If users wanted to open the raw file/download it, I would assume they would either explicitly download it  and/or **open with editor** to get the raw view

**Comment by jeffyjefflabs** [comment]
I agree with the other commenters that this behavior is confusing -- in particular because all the other actions starting with "Open" mean "inside the Lab interface" and this one is right next to them in the menu.  I like @jasongrout's idea of emphasizing "raw" but would suggest maybe "View Raw" instead of "Open Raw."  I would also suggest moving it down in the menu next to Download since the behavior is much more similar to that.

**Comment by isaacto** [comment]
I'd say it is "Download and View in Browser" (no need to mention "tab", it is the norm).  It should be a menu entry after "Download", not "Open with".  Also, the function intended by the one submitting this bug originally is useful.  I.e., I hope that I can easily open a new workspace containing only a particular notebook shown in the current file browser.  The reason is similar to my desire to open a URL in a new window when browsing the web.  That way I can partially overlap the display of the files, optimizing the usage of the screen estate.
